### PR TITLE
Android notification plugin

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds the Android notification plugin to handle lnurlpay_info, lnurlpay_invoice and swap_updated push notifications.

Builds on PR #410

### CODE REVIEW

**Review by roeierez — COMMENTED** [review]
Looks very good! Added a small comment and I think we should wait for https://github.com/breez/breez-sdk-liquid/pull/414 before merging as it will change slightly the receive interface

File: lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+            getNotificationManager(context)?.also { manager ->
+                createNotificationChannelGroups(context, manager)
+                createNotificationChannels(context, manager)
+                Log.d(TAG, "Registered notification channels")
+            }
+        }
+
+        @RequiresApi(Build.VERSION_CODES.O)
+        private fun createNotificationChannels(
```
Comment (erdemyerebasmaz) [inline lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt:134]: Reviewing this again made me aware of parts we could improve. I really want to refactor this function and make it more modular :)

Which poses a question: How can we ensure changes made in this plugin are also reflected in the GL notification plugin, and vice versa?

**Review by erdemyerebasmaz — COMMENTED** [review]
Changes look good to me!

I agree with @roeierez's sentiment to wait for https://github.com/breez/breez-sdk-liquid/pull/414 before merging, as it has breaking changes on how to address his suggestion:
> Should we compare to the notified swap id before notifying success?

File: lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt (hunk 1 above)
Comment (dangeross) [reply lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt:134]: I agree, there is overlap. I think we agreed in the dev call to keep it separate to begin with, then work together a more modular approach. I would suggest a separate repo for the core plugin components

Comment (roeierez) [reply lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/NotificationHelper.kt:134]: Let's follow our decision to first implement a workable solution then we will better understand on how to proceed.

**Review by roeierez — APPROVED** [review]
LGTM

**Review by erdemyerebasmaz — APPROVED** [review]

File: lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/job/SwapUpdated.kt [hunk 2]
```diff
@@ -49,7 +50,7 @@ class SwapUpdatedJob(
             is SdkEvent.PaymentSucceeded -> {
                 val payment = e.details
 
-                payment.swapId?.let { swapId ->
+                (payment.details as? PaymentDetails.Bitcoin)?.swapId?.let { swapId ->
```
Comment (dangeross) [inline lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/job/SwapUpdated.kt:53]: This should handle both `Lightning` and `Bitcoin` payment details

**Pull request merged**